mesonbuild/compilers.py

```diff
@@ -430,6 +430,17 @@ def _can_dedup(cls, arg):
         to recursively search for symbols in the libraries. This is not needed
         with other linkers.
         '''
+
+        # A Standalone -D must never be deduplicated because it is defined
+        # by what comes _after_ it. Thus dedupping this:
+        # -D FOO -D BAR
+        # would yield either
+        # -D FOO BAR
+        # or
+        # FOO -D BAR
+        # both of which are invalid.
+        if arg == '-D' or arg == '/D':
+            return 0
```

Comment:
Or, make a new list called `nodedup_args` and add -D, -I, -L, etc to it because those can also be used in this manner, fwict.